clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/CopybookDownloaderForE4E.ts

```diff
@@ -250,29 +255,66 @@ export class CopybookDownloaderForE4E {
     return false;
   }
 
-  private static getCopybookPath(
-    instance: string,
+  private static async getCopybookPath(
+    instance: string[],
     mapped: string,
     downloadFolder: string,
     copybook: string,
-  ): string {
+  ): Promise<string> {
     let folder = CopybookURI.createDatasetPath(
       instance,
       mapped,
       downloadFolder,
       E4E_FOLDER,
     );
 
-    if (!fs.existsSync(folder)) fs.mkdirSync(folder, { recursive: true });
+    /**
+     * There is an issue with VSCode File Watcher on Linux where it
+     * fails to watch subfolders changes when more subfolders are created
+     * all at once.
+     * https://github.com/microsoft/vscode/issues/142694
+     *
+     * As a workaround, the path is splitted into individual subfolders
+     * are they are created incrementally one by one.
+     */
+    const subdirectories = CopybookURI.createDatasetSubdirectories(
+      instance,
+      E4E_FOLDER,
+      mapped,
+    );
+    let finishedPath = downloadFolder;
+    for (const subdirectory of subdirectories) {
+      finishedPath = vscode.Uri.joinPath(
+        vscode.Uri.file(finishedPath),
+        subdirectory,
+      ).fsPath;
+
+      try {
+        await vscode.workspace.fs.createDirectory(
+          vscode.Uri.file(finishedPath),
+        );
```

Comment:
```suggestion
    let finishedPath = vscode.Uri.file(downloadFolder);
    for (const subdirectory of subdirectories) {
      finishedPath = vscode.Uri.joinPath(finishedPath, subdirectory);

      try {
        await vscode.workspace.fs.createDirectory(finishedPath);
```
and similarly the one join after the loop?